File: app/src/main/java/com/example/skogs/wifictrl/fragment/HotspotFragment.java
package com.example.skogs.wifictrl.fragment;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.example.skogs.wifictrl.R;
import com.example.skogs.wifictrl.adapter.HotspotAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by skogs on 20.04.2016.
 */
public class HotspotFragment extends AbstractTabFragment {

    private static final int LAYOUT = R.layout.fragment_hotspots;

    public static HotspotFragment getInstance(Context context){
        Bundle args = new Bundle();
        HotspotFragment fragment = new HotspotFragment();
        fragment.setArguments(args);
        fragment.setContext(context);
        fragment.setTitle(context.getString(R.string.tab_hotspots));

        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        view = inflater.inflate(LAYOUT, container, false);

        RecyclerView rv = (RecyclerView) view.findViewById(R.id.recyclerView);
        rv.setLayoutManager(new LinearLayoutManager(context));
        rv.setAdapter(new HotspotAdapter(createMockData()));

        return view;
    }

    private List<String> createMockData() {

        List<String> data = new ArrayList<>();
        data.add("Item 1");
        data.add("Item 1");
        data.add("Item 1");
        data.add("Item 1");
        data.add("Item 1");
        data.add("Item 1");

        return data;
    }

    public void setContext(Context context) {
        this.context = context;
    }
}

File: app/src/main/kotlin/com/example/skogs/wifictrl/fragment/HotspotFragment.kt
package com.example.skogs.wifictrl.fragment

import android.content.Context
import android.os.Bundle
import com.example.skogs.wifictrl.R
import java.util.*

/**
 * Created by skogs on 20.04.2016.
 */
class HotspotFragment : AbstractTabFragment() {

    override val layoutId: Int = R.layout.fragment_hotspots

    companion object {

        fun getInstance(context: Context): HotspotFragment {
            val args = Bundle()
            val fragment = HotspotFragment()
            fragment.arguments = args
            fragment.context = context
            fragment.title = context.getString(R.string.hotspots)

            return fragment
        }
    }

    /*override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        viewPage = inflater!!.inflate(layoutId, container, false)

        val rv = viewPage!!.findViewById(R.id.wifi_list) as RecyclerView
        rv.layoutManager = LinearLayoutManager(context)
        rv.adapter = HotspotAdapter(createMockData())

        return viewPage
    }*/

    private fun createMockData(): List<String> {

        val data = ArrayList<String>()
        data.add("Item 1")
        data.add("Item 1")
        data.add("Item 1")
        data.add("Item 1")
        data.add("Item 1")
        data.add("Item 1")

        return data
    }
}
